pig/src/main/java/org/jboss/pnc/bacon/pig/impl/config/BuildConfig.java

```diff
@@ -271,8 +269,9 @@ public String getEnvironmentId() {
             } catch (RemoteResourceException e) {
                 throw new FatalException("Exception while talking to PNC", e);
             }
-        } else {
-            throw new FatalException("No environmentId / environmentSystemImageId defined for the build config");
         }
+
+        // if we're here, no environmentId could be found
+        return null;
```

Comment:
Are you returning `null` from a method that returns `boolean`? Do you mean to return `false`? For fields, if you're using `Boolean` instead of `boolean`, you should probably validate the field with `@NotNull`.

Reply:
Oh shoot good catch. I didn't notice that I added this in the wrong area. THanks for pointing this out.

Reply:
@dwalluck I relooked at the code and it's Github that is confusing you into thinking this is for method `isUpgradableFrom`. It's for the `getEnvironmentId` that I'm returning null.

Reply:
Oops! I even double-checked and the line numbers seemed to match up. But I guess decide if you want to return `null` or throw an exception, in any case. I know old Java APIs like `File` used to return `null` and then they supposedly stopped doing that?

Reply:
Yeah. I wanted to throw NPE but I think that:
1. This won't actually happen due to the validation beforehand
2. I'll probably have to refactor the `getEnvironmentId` method because it does a bit too much, doesn't it?

I'm still on the fence on what to do with this. Any advice welcome! :)